dali/operators/audio/preemphasis_filter_op.h

```diff
@@ -31,47 +38,55 @@ const int kNumOutputs = 1;
 template<typename Backend>
 class PreemphasisFilter : public Operator<Backend> {
  public:
-  ~PreemphasisFilter() override = default;
+  explicit PreemphasisFilter(const OpSpec &spec)
+      : Operator<Backend>(spec), output_type_(spec.GetArgument<DALIDataType>(arg_names::kDtype)) {}
 
+  ~PreemphasisFilter() override = default;
   DISABLE_COPY_MOVE_ASSIGN(PreemphasisFilter);
 
- protected:
-  explicit PreemphasisFilter(const OpSpec &spec) :
-          Operator<Backend>(spec),
-          output_type_(spec.GetArgument<std::remove_const_t<decltype(this->output_type_)>>(
-                  arg_names::kDtype)) {}
+  bool CanInferOutputs() const override {
+    return true;
+  }
 
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    AcquireArguments(ws);
+    output_desc.resize(detail::kNumOutputs);
+    auto shape = input.shape();
+    output_desc[0].shape = shape;
+    output_desc[0].type = TypeTable::GetTypeInfo(output_type_);
+
+    if (std::is_same<CPUBackend, Backend>::value) {
+      assert(sample_queue_.empty());
+      for (int sample_id = 0; sample_id < shape.num_samples(); sample_id++) {
+        sample_queue_.push({volume(shape[sample_id]), sample_id});
+      }
```

Comment:
Nitpick: I find this queue less readable than having a vector and sorting it by volume.